### PR TITLE
[JAV-585]Loadbalance rule dynamic configuration is not work at some time

### PR DESCRIPTION
**Pull request by laijianbin**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://https://servicecomb.atlassian.net/projects/JAV/issues) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[JAV-XXX] Fixes bug in ApproximateQuantiles`, where you replace `JAV-XXX` with the appropriate JIRA issue.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean install` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

File: handlers/handler-loadbalance/src/main/java/io/servicecomb/loadbalance/LoadbalanceHandler.java [hunk 1]
```diff
@@ -92,9 +92,9 @@ public LoadbalanceHandler() {
   public void handle(Invocation invocation, AsyncResponse asyncResp) throws Exception {
     String policy = Configuration.INSTANCE.getPolicy(invocation.getMicroserviceName());
     String strategy = Configuration.INSTANCE.getRuleStrategyName(invocation.getMicroserviceName());
-    boolean isRuleNotChanged = StringUtils.equals(policy,this.policy) && StringUtils.equals(strategy, this.strategy);
-    
-    if (!isRuleNotChanged){
+    boolean isRuleNotChanged = isEqual(policy, this.policy) && isEqual(strategy,this.strategy);
```
Comment (wujimin) [inline handlers/handler-loadbalance/src/main/java/io/servicecomb/loadbalance/LoadbalanceHandler.java]: not format code?

File: handlers/handler-loadbalance/src/test/java/io/servicecomb/loadbalance/TestLoadbalanceHandler.java [hunk 2]
```diff
@@ -437,4 +437,19 @@ public void sendWithRetry() {
...
+  public void testIsEqual(){
+    boolean nullResult = handler.isEqual(null, null);
+    Assert.assertEquals(true, nullResult);  
+    boolean bothNotNullResult = handler.isEqual("com.netflix.loadbalancer.RandomRule", "com.netflix.loadbalancer.RandomRule");
+    Assert.assertEquals(true, bothNotNullResult);  
+    boolean globalNotNull = handler.isEqual(null, "com.netflix.loadbalancer.RandomRule");
+    Assert.assertEquals(false, globalNotNull);  
+    boolean localNotNull = handler.isEqual("com.netflix.loadbalancer.RandomRule", null);
+    Assert.assertEquals(false, localNotNull);  
+
```
Comment (wujimin) [inline handlers/handler-loadbalance/src/test/java/io/servicecomb/loadbalance/TestLoadbalanceHandler.java]: delete empty lines

File: handlers/handler-loadbalance/src/main/java/io/servicecomb/loadbalance/LoadbalanceHandler.java (hunk 1 above)
Comment (laijianbin) [reply handlers/handler-loadbalance/src/main/java/io/servicecomb/loadbalance/LoadbalanceHandler.java]: Yeah, I forget it and I  have modified

File: handlers/handler-loadbalance/src/test/java/io/servicecomb/loadbalance/TestLoadbalanceHandler.java (hunk 2 above)
Comment (laijianbin) [reply handlers/handler-loadbalance/src/test/java/io/servicecomb/loadbalance/TestLoadbalanceHandler.java]: okay, and I finished

**Comment by WillemJiang** [comment]
@laijianbin  I don't think you logged a right JIRA ID. Can you double check the JIRA ID?

**Comment by weichao666** [comment]
@WillemJiang  The JIRA ID comes from commerce  JIRA and you means create an issue at opensource JIRA?

**Comment by laijianbin** [comment]
@WillemJiang create an task at opensource JIRA and use opensource JIRA ID?

**Comment by laijianbin** [comment]
@WillemJiang create a task at opensource JIRA and use opensource JIRA ID?

**Pull request merged**